### PR TITLE
Add support for configuring secrets

### LINKED ISSUE #50: Add support for image pull secrets
## Context
To verify the signature of images which are part of a private registry, the corresponding image pull secret needs to exist in the same namespace (see: https://kyverno.io/docs/writing-policies/verify-images/#using-private-registries).

The necessary additional argument (`--imagePullSecrets=name-of-the-secret`) can already be configured through the `extraArgs` parameter.

## Alternatives
Manually managing the image pull secrets would be the undesired alternative.

### PR DESCRIPTION
**Pull request by corvus-ch**
Resolves #50.

## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [x] Link this PR to related issues or PRs.

### CODE REVIEW

File: tests/golden/defaults/kyverno/kyverno/10_secrets.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
+data: {}
```
Comment (corvus-ch) [inline tests/golden/defaults/kyverno/kyverno/10_secrets.yaml:2]: Does it work when there are both `data` and `stringData` are present?

File: tests/golden/defaults/kyverno/kyverno/10_secrets.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+kind: Secret
+metadata:
+  annotations: {}
+  labels:
+    name: myregistry
+  name: myregistry
+stringData:
+  .dockerconfigjson: t-my-tenant/c-my-cluster/kyverno/image_pullsecret
```
Comment (corvus-ch) [inline tests/golden/defaults/kyverno/kyverno/10_secrets.yaml:10]: I lost `?{…}`, why?

Comment (corvus-ch) [reply tests/golden/defaults/kyverno/kyverno/10_secrets.yaml:10]: This is to be expected in catalog compile.

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 3]
```diff
@@ -166,3 +166,20 @@ type:: string
...
++
+[source,yaml]
+----
+myregistry:
+  type: kubernetes.io/dockerconfigjson
+  stringData:
+    .dockerconfigjson: "?{vaultkv:t-my-tenant/c-my-cluster/kyverno/image_pullsecret}"
+----
+
+Allows to configure arbitrary secrets.
```
Comment (simu) [inline docs/modules/ROOT/pages/references/parameters.adoc:185]:
```suggestion
This parameter allows users to deploy arbitrary secrets.
Each entry is transformed into a Secret resource.
The key is used as the name of the resulting resource.
The provided value is merged with an empty Secret resource.
The component doesn't validate the provided secret configurations.

Users can remove secrets configured higher-up in the hierarchy by setting the corresponding value to `null`.

IMPORTANT: Always use `stringData` when using Vault references in secret configurations.
```

**Review by simu — APPROVED** [review]

**Review by anothertobi — APPROVED** [review]

**Pull request merged**